server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java

```diff
@@ -39,6 +40,7 @@ public CleanupTasksPerformer(Set<CleanupTask> cleanupTasks) {
         this.cleanupTasks = cleanupTasks;
     }
 
+    @PreDestroy
```

Comment:
Shouldn't it be done on a per-task basis?

Reply:
There are only classes that are not injected anywhere so my reasoning is that they are created after their dependencies (think, Cassandra cleanup task requires Cassandra so will be created after Cassandra) and will then be called before application component without any ordering.
I think this is more than enough and we don't need to change the registration process of every cleanup tasks.
WDYT ?

Reply:
I am not sure a single point where all cleanUp task can be called safely...

Reply:
what do you mean by safely ?